src/pages/dar_application/DataUseAgreements.jsx

```diff
@@ -14,11 +13,77 @@ export default function DataUseAgreements(props) {
     darCode,
     isAttested,
     cancelAttest,
+    datasets
   } = props;
+  const [daas, setDaas] = useState([]);
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        const daaList = await DAA.getDaas();
+        setDaas(daaList);
+      } catch (error) {
+        Notifications.showError({
+          text: 'Error: Unable to retrieve DAAs from server',
+        });
+      }
+    };
+    init();
+  }, []);
+
+  const RequiredDAAs = () => {
+    const fileNames = new Set();
+    const daaDivs = datasets.map((dataset) => {
+      const datasetDacId = dataset.dacId;
+      if (!datasetDacId) {
+        return <div key={dataset.id}></div>;
+      }
+      const daa = daas.find((daa) => daa.dacs.some((d) => d.dacId === datasetDacId));
+      const id = daa.daaId;
+      const fileName = daa.file.fileName.split('.')[0];
+      if (fileNames.has(fileName)) {
+        return <div key={id-dataset.name}></div>;
+      }
+      fileNames.add(fileName);
+      return (
+        <div key={id}>
+          {DAADownload(id, fileName)}
+        </div>
+      );
+    });
+    if (fileNames.size === 0) {
+      return (
+        <div></div>
```

Comment:
In the empty case, should we default to the original `BroadLibraryCardAgreementLink` and `NhgriLibraryCardAgreementLink` components? I think we should clarify the requirements here.

Reply:
Ideally, we should not be running into this empty case, because we should only be opening DARs for datasets that are associated with DACS (not Open Access or External to DUOS), and thus that have DAAs. I think currently until we have a fix in for that in, in the empty case we should just be showing the DUOS "DAA"/Code of Conduct that is displayed in text here. What do you think?

Reply:
> Ideally, we should not be running into this empty case, because we should only be opening DARs for datasets that are associated with DACS (not Open Access or External to DUOS), and thus that have DAAs. I think currently until we have a fix in for that in, in the empty case we should just be showing the DUOS "DAA"/Code of Conduct that is displayed in text here. What do you think?

That is the ideal case. I think we still need a way to deal with the edge case which is when a DAC does not have a DAA associated to it. In testing on dev, I was able to create a DAR against an open access dataset. We should be protecting against those cases, but they inevitably slip through.

Reply:
Update: We have decided that an empty div and the displaying of the DUOS Code of Conduct makes sense to show in these cases where a DAR is created against an open access or external dataset.